src/ast.rs

```diff
@@ -6223,6 +6233,26 @@ impl ClosureTypeName {
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct LambdaSig(Vec<TypeHandle>);
 
+impl LambdaSig {
+    fn demangle_inner<'subs, 'prev, 'ctx, W>(
```

Comment:
It isn't clear to me why this was pulled out into a helper function, but it doesn't really matter. I do think that we should not be calling this `demangle_inner`, though. It is a bit too much like the existing `demangle_as_inner` trait method, and could be confused with that.

Reply:
It was pulled into a helper function because it contains early returns via `?`, which if done naively would leave `ctx.is_lambda_arg` in a bad state. Do you have a better name suggestion?

Reply:
`demangle_args`?

Reply:
Done.